internal/store/store.go

```diff
@@ -7,10 +7,62 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/lib/pq"
+
 	// Register lib/pq PostreSQL driver
 	_ "github.com/lib/pq"
 )
 
+// A NoResultError indicates that no data matching the query was found.
+type NoResultError struct {
+	err error
+}
+
+func (nre NoResultError) Error() string {
+	return fmt.Sprintf("no results found: %v\n", nre.err)
+}
+
+type postgresError struct {
```

Comment:
I'm not really sure we need this type at all. We should expect that the query is written right. Really the only errors we care about or expect are like: it failed, or it returned no results.